src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -503,6 +513,40 @@ private static void createBranches(List<LfBus> lfBuses, LfNetwork lfNetwork, LfT
         }
     }
 
+    private static void updateControlArea(Bus bus, LfBus lfBus, LfNetwork network, LfNetworkParameters parameters, LoadingContext loadingContext, LfNetworkLoadingReport report) {
+        Optional<Area> areaOpt = bus.getVoltageLevel().getArea(parameters.getAreaInterchangeControlAreaType());
+        areaOpt.ifPresent(area -> {
+            LfControlArea controlArea = network.getControlAreaById(area.getId());
+            if (controlArea == null) {
+                controlArea = createControlArea(area, parameters, loadingContext, network);
+            }
+            controlArea.addBus(lfBus);
+        });
+    }
+
+    private static LfControlArea createControlArea(Area area, LfNetworkParameters parameters, LoadingContext loadingContext, LfNetwork network) {
+        LfControlArea lfControlArea = LfControlAreaImpl.create(area, network, parameters);
+        network.addControlArea(lfControlArea);
+        area.getAreaBoundaryStream().forEach(areaBoundary -> {
+            if (areaBoundary.getTerminal().isPresent()) {
+                Terminal terminal = areaBoundary.getTerminal().get();
+                loadingContext.controlAreaBoundaries.put(terminal, lfControlArea);
+            }
+            if (areaBoundary.getBoundary().isPresent()) {
+                DanglingLine danglingLine = areaBoundary.getBoundary().get().getDanglingLine();
+                loadingContext.controlAreaBoundaries.put(danglingLine.getTerminal(), lfControlArea);
```

Comment:
Can we really "just" consider the Boundary terminal? @vmouradian : please check if this is the right way to do it

Reply:
Discussed together: the map name was confusing but you are not adding boundaries in this map, just terminals which enable you to detect and add the boundaries later